quic/s2n-quic-transport/src/stream/manager.rs

```diff
@@ -427,9 +427,132 @@ pub struct AbstractStreamManager<S> {
 #[allow(unknown_lints, clippy::non_send_fields_in_send_ty)]
 unsafe impl<S> Send for AbstractStreamManager<S> {}
 
-impl<S: StreamTrait> AbstractStreamManager<S> {
-    /// Creates a new `StreamManager` using the provided configuration parameters
-    pub fn new(
+impl<S: 'static + StreamTrait> AbstractStreamManager<S> {
```

Comment:
What distinguishes functions that end up in the AbstractStreamManager versus functions that end up in the stream::Manager?

Reply:
The ones in stream::Manager are the ones called by the connection. Each one is current required to be implemented to drive streams. Outside of that, the other functions are private to the AbstractStreamManager and aren't required to hold a particular contract.